harness/determined/common/experimental/experiment.py

```diff
@@ -514,10 +516,12 @@ def get_with_offset(offset: int) -> bindings.v1GetExperimentCheckpointsResponse:
                 states=[bindings.checkpointv1State.COMPLETED],
             )
 
-        resps = api.read_paginated(
-            get_with_offset=get_with_offset,
-            pages=api.PageOpts.single if max_results else api.PageOpts.all,
-        )
+        with self._session as sess:
+            resps = api.read_paginated(
```

Comment:
This is still one of my least-favorite parts of the SDK. Would a future design want to build this into `BaseSession`?

Reply:
i don't love it either, but i dunno. could also see `read_paginated` belonging somewhere in `bindings`. definitely out of scope for this PR